engine/src/main/java/org/terasology/rendering/dag/StateManager.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+
+import com.google.api.client.util.Maps;
+import java.util.Collection;
+import java.util.Map;
+import org.terasology.context.Context;
+import org.terasology.rendering.dag.states.State;
+import org.terasology.rendering.dag.states.StateType;
+import org.terasology.rendering.dag.states.StateTypeImpl;
+import org.terasology.rendering.dag.states.StateValue;
+import org.terasology.rendering.dag.states.Wireframe;
+
+/**
+ * TODO: add javadocs
+ */
+public class StateManager {
```

Comment:
In line with previous comments this architecture doesn't work for me, sorry. We need a RenderGraph, a RenderPipelineGenerator and a RenderPipelineOptimizer. To start with it would be good to focus only on the first two. We can think about the last one once we have nailed the first two.